Feature/ects calculator

SharpP03 opened the pull request.

SharpP03 committed:
Service for fetching list of subjects.

SharpP03 committed:
Subject selection refactor (fetch).

SharpP03 committed:
AsyncStorage integration for saving calculator items.

SharpP03 committed:
in progress: Add subject editing functionality.

SharpP03 committed:
Error messages language support (localization).
+ summary menu display correction.

SharpP03 committed:
Outsource CalculatorScreen functions: integrate SubjectPopup as new component.

SharpP03 committed:
Update Polish translation for course removal button text.

SharpP03 committed:
Add posthog-react-native dependency and update react-native-localize version.

SharpP03 committed:
Merge branch 'main' into feature/ECTScalculator

SharpP03 committed:
Refactor CalculatorScreen and SubjectPopup to use theme-based styles;

SharpP03 committed:
integration with main in progrss: Refactor styles and localization in CalculatorScreen and DropdownMenu components.

SharpP03 committed:
Styles fix for subject grade list.

SharpP03 committed:
Merge branch 'main' into feature/ECTScalculator

SharpP03 committed:
style  fix: delete button and dropdown menu.

SharpP03 committed:
Fix selected item button style.

SharpP03 committed:
Posthog deletion.

Ereffe commented on src/app/tabs/CalculatorScreen/SubjectPopup.tsx:
```diff
@@ -0,0 +1,153 @@
+type Props = {
+  subjectName: string;
+  setSubjectName: (value: string) => void;
+  ectsPoints: string;
```
```suggestion
  Group the props, to maintain readability and sustainability
```

Ereffe commented on src/app/tabs/CalculatorScreen/SubjectPopup.tsx:
```diff
@@ -0,0 +1,153 @@
+          ref={gradeInput}
+          style={
+            gradeError && gradeFocused
+              ? styles.userInputFocusedError
```
same as above

SharpP03 committed:
Calculator ready for modification.

florczaq committed:
Refactor CalculatorScreen: Remove old styles, introduce new components and styles

- Deleted CalculatorStyles.ts and moved styles to individual components.
- Added new components: BatchActions, HeaderRow, ItemRow, PopupForm, SubjectsList, and SummaryPanel.
- Created corresponding styles for each new component.
- Introduced a new colors utility for CalculatorScreen styles.
- Defined CalcItem type for better type safety.

florczaq committed:
Refactor CalculatorScreen and PopupForm: Improve state management, enhance error handling, and optimize performance with memoization

SharpP03 committed:
Refactor CalculatorScreen and related components: change layout with SummaryPanel, improve selection handling in HeaderRow and ItemRow, and update styles.

SharpP03 committed:
Minor fix - header theme colors.

SharpP03 committed:
refactor style array to table with overrides.

SharpP03 committed:
Refactor PopupForm and CalculatorScreen: Enhance prop management for readability and maintainability by grouping related props and simplifying state handling.

Ereffe commented on src/app/tabs/CalculatorScreen/components/PopupForm.tsx:
```diff
@@ -0,0 +1,190 @@
+// ...existing code...
```
... please delete this comment XD

Ereffe commented on src/app/tabs/CalculatorScreen/components/PopupForm.tsx:
```diff
@@ -0,0 +1,190 @@
+};
+
+export default PopupForm;
+// ...existing code...
```
Same, del this

Ereffe committed:
Remove unnecessary comments in PopupForm.tsx

Ereffe committed:
Merge branch 'main' into feature/ECTScalculator
